src/main/java/com/cloudbees/jenkins/plugins/BitbucketJobProbe.java

```diff
@@ -65,6 +68,15 @@ public void triggerMatchingJobs(String user, String url, String scm) {
         }
     }
 
+    private boolean hasBeenTriggered(List<SCM> scmTriggered, SCM scmTrigger) {
+        for (SCM scm : scmTriggered) {
+            if (scm.equals(scmTrigger)) {
```

Comment:
Well, I think this fix is ok as far as all SCM implementations override `equals`, so downtream PRs would be needed for this.
Actually, this fix seems to be fixing  the original issue (probably collaterally and not covering all the scenarios), but it's better than nothing.

My +1 is for keeping this PR merged and start follow up PRs (or at least create JIRA tickets) to override `equals` in all known `SCM` implementations (which I think will be required for JENKINS-29326 anyway).